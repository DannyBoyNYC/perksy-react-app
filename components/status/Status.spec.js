const React = require('react');
const { shallow } = require('enzyme');

const Status = require('./index').default;

describe('<Status.* />', () => {
  let component;
  const requiredProps = {
    code: 'draft'
  };

  describe('<Status.Button />', () => {
    beforeEach(() => {
      component = shallow(<Status.Button {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('status status--button');
    });
  });
});
