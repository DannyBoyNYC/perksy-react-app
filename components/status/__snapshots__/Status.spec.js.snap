// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status.* /> <Status.Button /> should render correctly with all required props 1`] = `
<div
  className="status status--button"
>
  <span
    className="status__code"
  >
    Draft
  </span>
</div>
`;
